datalad/distribution/create_sibling_github.py

```diff
@@ -0,0 +1,373 @@
+        ds = require_dataset(
+            dataset, check_installed=True, purpose='create Github sibling')
+        # gather datasets and essential info
+        # dataset instance and mountpoint relative to the top
+        toprocess = [(ds, '')]
+        if recursive:
+            for d in ds.get_subdatasets(
+                    fulfilled=None,  # we want to report on missing dataset in here
+                    absolute=False,
+                    recursive=recursive,
+                    recursion_limit=recursion_limit):
+                sub = Dataset(opj(ds.path, d))
+                if not sub.is_installed():
+                    lgr.info('Ignoring unavailable subdataset %s', sub)
+                    continue
+                toprocess.append((sub, d))
+
+        # check for existing remote configuration
+        filtered = []
+        for d, mp in toprocess:
+            if sibling_name in d.repo.get_remotes():
+                if existing == 'error':
+                    msg = '{} already had a configured sibling "{}"'.format(
+                        d, sibling_name)
+                    if dryrun:
+                        lgr.error(msg)
+                    else:
+                        raise ValueError(msg)
+                elif existing == 'skip':
+                    continue
+            gh_reponame = '{}{}{}'.format(
+                reponame,
+                '-' if mp else '',
+                template_fx(mp))
+            filtered.append((d, gh_reponame))
+
+        if not filtered:
+            # all skipped
+            return []
+
+        # actually make it happen on Github
+        rinfo = _make_github_repos(
+            gh, github_user, github_passwd, github_organization, filtered,
+            existing, access_protocol, dryrun)
+
+        # lastly configure the local datasets
+        for d, url, existed in rinfo:
+            if not dryrun:
+                # first make sure that annex doesn't touch this one
+                # but respect any existing config
+                ignore_var = 'remote.{}.annex-ignore'.format(sibling_name)
+                if not ignore_var in d.config:
+                    d.config.add(ignore_var, 'true', where='local')
+                AddSibling()(
+                    dataset=d,
+                    name=sibling_name,
+                    url=url,
+                    recursive=False,
+                    # TODO fetch=True, maybe only if one existed already
+                    force=existing in {'reconfigure'},
+                    publish_depends=publish_depends)
+
+        # TODO let submodule URLs point to Github (optional)
+        return rinfo
+
+    @staticmethod
+    def result_renderer_cmdline(res, args):
+        from datalad.ui import ui
+        res = assure_list(res)
+        if args.dryrun:
+            ui.message('DRYRUN -- Anticipated results:')
+        if not len(res):
+            ui.message("Nothing done")
+        else:
+            for d, url, existed in res:
```

Comment:
I think output is nowhere described... Btw shouldn't it be an ordered (if order matters) dict with d as a key?  I guess I would be to look also into regular create-sibling..  beasts should be similar